src/platform/cmake/FindLuaJIT.cmake

```diff
@@ -0,0 +1,225 @@
+    set(LUAJIT_VERSION_STRING "${LUAJIT_VERSION_MAJOR}.${LUAJIT_VERSION_MINOR}.${LUAJIT_VERSION_PATCH}")
+  else ()
+    string(REGEX REPLACE ".*;#define[ \t]+LUAJIT_VERSION[ \t]+\"LuaJIT ([0-9.]+[^\"]*)\"[ \t]*;.*" "\\1" LUAJIT_VERSION_STRING ";${luajit_version_strings};")
+    string(REGEX REPLACE "^([0-9]+)\\.[^\"]*$" "\\1" LUAJIT_VERSION_MAJOR "${LUAJIT_VERSION_STRING}")
+    string(REGEX REPLACE "^[0-9]+\\.([0-9]+)[^\"]*$" "\\1" LUAJIT_VERSION_MINOR "${LUAJIT_VERSION_STRING}")
+    string(REGEX REPLACE "^[0-9]+\\.[0-9]+\\.([0-9]).*" "\\1" LUAJIT_VERSION_PATCH "${LUAJIT_VERSION_STRING}")
+  endif ()
+  foreach (ver IN LISTS _luajit_append_versions)
+    if (ver STREQUAL "${LUAJIT_VERSION_MAJOR}.${LUAJIT_VERSION_MINOR}")
+      set(LUAJIT_VERSION_MAJOR ${LUAJIT_VERSION_MAJOR} PARENT_SCOPE)
+      set(LUAJIT_VERSION_MINOR ${LUAJIT_VERSION_MINOR} PARENT_SCOPE)
+      set(LUAJIT_VERSION_PATCH ${LUAJIT_VERSION_PATCH} PARENT_SCOPE)
+      set(LUAJIT_VERSION_STRING ${LUAJIT_VERSION_STRING} PARENT_SCOPE)
+      return()
+    endif ()
+  endforeach ()
+endfunction(_luajit_get_header_version)
+
+function(_luajit_find_header)
+  _luajit_set_version_vars()
+
+  # Initialize as local variable
+  set(CMAKE_IGNORE_PATH ${CMAKE_IGNORE_PATH})
+  while (TRUE)
+    # Find the next header to test. Check each possible subdir in order
+    # This prefers e.g. higher versions as they are earlier in the list
+    # It is also consistent with previous versions of FindLua
+    foreach (subdir IN LISTS _luajit_include_subdirs)
+      find_path(LUAJIT_INCLUDE_DIR luajit.h
+        HINTS ENV LUAJIT_DIR
+        PATH_SUFFIXES ${subdir}
+        )
+      if (LUAJIT_INCLUDE_DIR)
+        break()
+      endif()
+    endforeach()
+    # Did not found header -> Fail
+    if (NOT LUAJIT_INCLUDE_DIR)
+      return()
+    endif()
+    _luajit_get_header_version()
+    # Found accepted version -> Ok
+    if (LUAJIT_VERSION_STRING)
+      if (LUAJIT_Debug)
+        message(STATUS "Found suitable version ${LUAJIT_VERSION_STRING} in ${LUAJIT_INCLUDE_DIR}/lua.h")
+      endif()
+      return()
+    endif()
+    # Found wrong version -> Ignore this path and retry
+    if (LUAJIT_Debug)
+      message(STATUS "Ignoring unsuitable version in ${LUAJIT_INCLUDE_DIR}")
+    endif()
+    list(APPEND CMAKE_IGNORE_PATH "${LUAJIT_INCLUDE_DIR}")
+    unset(LUAJIT_INCLUDE_DIR CACHE)
+    unset(LUAJIT_INCLUDE_DIR)
+    unset(LUAJIT_INCLUDE_DIR PARENT_SCOPE)
+  endwhile ()
+endfunction()
+
+_luajit_get_versions()
+_luajit_find_header()
+_luajit_get_header_version()
+unset(_luajit_append_versions)
+
+if (LUAJIT_VERSION_STRING)
+  set(_luajit_library_names
+    luajit${LUAJIT_VERSION_MAJOR}${LUAJIT_VERSION_MINOR}
+    luajit${LUAJIT_VERSION_MAJOR}.${LUAJIT_VERSION_MINOR}
+    luajit-${LUAJIT_VERSION_MAJOR}.${LUAJIT_VERSION_MINOR}
+    luajit.${LUAJIT_VERSION_MAJOR}.${LUAJIT_VERSION_MINOR}
+    lua51
```

Comment:
Does LuaJIT actually provide a `lua51` soname? That seems weird...

Reply:
Good point. That is an inconsistency.

I'm using a win64 docker and MinGW's LuaJIT has same library names as standard Lua 5.1. In unix that's not true, names would be something like luajit51, luajit5.1...

Reply:
Hm, if that's the standard soname that's getting shipped for the package in MSYS2, then I guess it should stay as it is.

Reply:
Well, I've added luajit51 variants in order to be able to build using unix systems. I don't know if I should left the luajit${LUAJIT_VERSION_MAJOR}${LUAJIT_VERSION_MINOR} variants into the list. I've found LuaJIT installations where the library was named something like libluajit2.1.so or something like that, but I supose those name conventions are customized. Should I remove ${LUAJIT_VERSION_MAJOR}${LUAJIT_VERSION_MINOR} variants from the list leaving only luajit51 and lua51 ones, or it's alrhigt like it is now? Currently possibly library names list has 12 items...